modules/perception/camera/app/lane_camera_perception.cc

```diff
@@ -0,0 +1,239 @@
+/******************************************************************************
+* Copyright 2019 The Apollo Authors. All Rights Reserved.
+*
+* Licensed under the Apache License, Version 2.0 (the License);
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+* http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an AS IS BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*****************************************************************************/
+#include "modules/perception/camera/app/lane_camera_perception.h"
+
+#include <gflags/gflags.h>
+#include <yaml-cpp/yaml.h>
+#include <algorithm>
+#include <fstream>
+#include <string>
+#include <utility>
+#include <vector>
+
+#include "cyber/common/log.h"
+#include "modules/common/util/file.h"
+#include "modules/perception/base/object.h"
+#include "modules/perception/camera/app/debug_info.h"
+#include "modules/perception/camera/common/global_config.h"
+#include "modules/perception/camera/common/util.h"
+#include "modules/perception/common/io/io_util.h"
+#include "modules/perception/inference/utils/cuda_util.h"
+#include "modules/perception/lib/utils/perf.h"
+
+namespace apollo {
+namespace perception {
+namespace camera {
+
+using apollo::common::util::GetAbsolutePath;
+
+bool LaneCameraPerception::Init(
+    const CameraPerceptionInitOptions &options) {
+  std::string work_root = "";
+  if (options.use_cyber_work_root) {
+    GetCyberWorkRoot(&work_root);
+  }
+  std::string config_file =
+      GetAbsolutePath(options.root_dir, options.conf_file);
+  config_file = GetAbsolutePath(work_root, config_file);
+  CHECK(apollo::common::util::GetProtoFromFile(
+        config_file, &perception_param_)) << "Read config failed: ";
+  CHECK(inference::CudaUtil::set_device_id(perception_param_.gpu_id()));
+
+  lane_calibration_working_sensor_name_ =
+      options.lane_calibration_working_sensor_name;
+
+  // InitLane
+  base::BaseCameraModelPtr model;
+  InitLane(work_root, model, perception_param_);
+
+  // Init calibration service
+  InitCalibrationService(work_root, model, perception_param_);
+
+  return true;
+}
+
+void LaneCameraPerception::InitLane(
+    const std::string &work_root,
+    base::BaseCameraModelPtr &model,
+    const app::PerceptionParam &perception_param) {
+  // Init lane
+  CHECK(perception_param.has_lane_param())
+  << "Failed to include lane_param";
```

Comment:
Would it fit in previous line? If not, please also leave 4 spaces before "<<".

Reply:
done